File: integerGAmatlab/utils/operators/selection.m
function [next_pop, next_fit] = selection(next_size, pop1, fit1, pop2, fit2)
    imm_pop = [pop1; pop2];
    imm_fit = [fit1; fit2];
    
    n = size(imm_pop, 1);
    idx = (1:n);
    
    for i=1:n-1
        for j=i+1:n
            if imm_fit(idx(i)) < imm_fit(idx(j))
                tmp = idx(i);
                idx(i) = idx(j);
                idx(j) = tmp;
            end
        end
    end
    
    next_pop = zeros(next_size, size(pop1, 2));
    next_fit = zeros(next_size, 1);
    for i=1:next_size
        next_pop(i, :) = imm_pop(idx(i), :);
        next_fit(i) = imm_fit(idx(i));
    end
end

File: integerGA_python/utils/operators/selection.py
"""
Selection operator for combining populations.
"""
import numpy as np


def selection(next_size, pop1, fit1, pop2, fit2):
    """
    Select best individuals from two populations.
    
    Parameters:
    -----------
    next_size : int
        Size of next population
    pop1 : ndarray
        First population
    fit1 : ndarray
        Fitness of first population
    pop2 : ndarray
        Second population
    fit2 : ndarray
        Fitness of second population
        
    Returns:
    --------
    tuple : (next_pop, next_fit) - Selected population and fitness
    """
    # Combine populations
    imm_pop = np.vstack([pop1, pop2])
    imm_fit = np.concatenate([fit1, fit2])
    
    # Sort by fitness (descending order - maximization)
    idx = np.argsort(imm_fit)[::-1]
    
    # Select top individuals
    next_pop = imm_pop[idx[:next_size]]
    next_fit = imm_fit[idx[:next_size]]
    
    return next_pop, next_fit
